tests/soak-tests/src/test/java/org/apache/activemq/artemis/tests/soak/owleak/OWLeakTest.java

```diff
@@ -0,0 +1,242 @@
+
+   public static final String SERVER_NAME_0 = "openwire-leaktest";
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+   private static final String TEST_NAME = "OW_LEAK";
+   private static final boolean TEST_ENABLED = Boolean.parseBoolean(testProperty(TEST_NAME, "TEST_ENABLED", "true"));
+   private static final String PROTOCOL_LIST = testProperty(TEST_NAME, "PROTOCOL_LIST", "OPENWIRE");
+   private final String protocol;
+   private final int NUMBER_OF_MESSAGES;
+   private final int PRODUCERS;
+   private final int MESSAGE_SIZE;
+   Process serverProcess;
+
+   public OWLeakTest(String protocol) {
+      this.protocol = protocol;
+      NUMBER_OF_MESSAGES = intMandatoryProperty(TEST_NAME, protocol + "_NUMBER_OF_MESSAGES");
+      PRODUCERS = intMandatoryProperty(TEST_NAME, protocol + "_PRODUCERS");
+      MESSAGE_SIZE = intMandatoryProperty(TEST_NAME, protocol + "_MESSAGE_SIZE");
+   }
+
+   @Parameterized.Parameters(name = "protocol={0}")
+   public static Collection<Object[]> parameters() {
+      String[] protocols = PROTOCOL_LIST.split(",");
+
+      ArrayList<Object[]> parameters = new ArrayList<>();
+      for (String str : protocols) {
+         logger.info("Adding {} to the list for the test", str);
+         parameters.add(new Object[]{str});
+      }
+
+      return parameters;
+   }
+
+   @Before
+   public void before() throws Exception {
+      Assume.assumeTrue(TEST_ENABLED);
+      cleanupData(SERVER_NAME_0);
+
+      serverProcess = startServer(SERVER_NAME_0, 0, 10_000);
+   }
+
+
+   public static void main(String[] arg) {
+      int PRODUCERS = Integer.parseInt(arg[0]);
+      int NUMBER_OF_MESSAGES = Integer.parseInt(arg[1]);
+      int MESSAGE_SIZE = Integer.parseInt(arg[2]);
+      String protocol = arg[3];
+      ExecutorService service = Executors.newFixedThreadPool(PRODUCERS + 1 + 1);
+
+      String QUEUE_NAME = "some_queue";
+
+      Semaphore semaphore = new Semaphore(PRODUCERS + 1);
+
+      CountDownLatch latch = new CountDownLatch(PRODUCERS + 1 + 1);
+
+      AtomicBoolean running = new AtomicBoolean(true);
+
+      AtomicInteger errors = new AtomicInteger(0);
+
+      try {
+
+         for (int i = 0; i < PRODUCERS; i++) {
+            final int t = i;
+            ConnectionFactory factory = CFUtil.createConnectionFactory(protocol, "tcp://localhost:61616");
+            service.execute(() -> {
+               try {
+                  for (int msg = 0; msg < NUMBER_OF_MESSAGES; msg++) {
+                     Connection connection = factory.createConnection();
+                     Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+                     MessageProducer producer = session.createProducer(session.createQueue(QUEUE_NAME));
+                     char[] msgStr = new char[MESSAGE_SIZE];
+                     Arrays.fill(msgStr, 'a');
```

Comment:
I think it would be better to use a much-less-quick-repeated sequence for the message bodies, to help validate what comes back are the messages expected, and they arent actually malformed in some way. E.g loop generating the alphabet, optionally complicate with inserting a count of the loops. Still easily generated and verified while not 100% oblivious to the payload being goosed.

Maybe also add a header to index each message, verify all the indexes are received and its not e.g the same message repeated.